Source/Renderer/GLSLModernizer.js

```diff
@@ -0,0 +1,233 @@
+define([
+        '../Core/defined',
+        '../Core/DeveloperError',
+    ], function(
+        defined,
+        DeveloperError) {
+    'use strict';
+
+    /**
+     * A function to port GLSL shaders from GLSL ES 1.00 to GLSL ES 3.00
+     *
+     * This function is nowhere near comprehensive or complete. It just
+     * handles some common cases.
+     *
+     * Note that this function requires the presence of the
+     * "#define OUTPUT_DECLARATION" line that is appended
+     * by ShaderSource.
+     *
+     * @private
+     */
+    function glslModernizeShaderText(source, isFragmentShader) {
+        var outputDeclarationRegex = /#define OUTPUT_DECLARATION/;
+        var splitSource = source.split('\n');
+
+        if (source.search(/#version 300 es/g) !== -1) {
+            return source;
+        }
+
+        var outputDeclarationLine = -1;
+        var number;
+        for (number = 0; number < splitSource.length; ++number) {
+            var line = splitSource[number];
+            if (outputDeclarationRegex.exec(line)) {
+                outputDeclarationLine = number;
+                break;
+            }
+        }
+
+        if (outputDeclarationLine === -1) {
+            throw new DeveloperError('Could not find a #define OUTPUT_DECLARATION nor a main function!');
+        }
+
+        // Note that this fails if your string looks like
+        // searchString[singleCharacter]searchString
+        function replaceInSource(str, replacement, strIsRegex) {
+            var regex = str;
+            if ((strIsRegex === undefined) || strIsRegex === false) {
+                var regexStr = '(^|[^\\w])(' + str + ')($|[^\\w])';
+                regex = new RegExp(regexStr, 'g');
+            }
+
+            for (var number = 0; number < splitSource.length; ++number) {
+                var line = splitSource[number];
+                if (strIsRegex) {
+                    splitSource[number] = line.replace(regex, replacement);
+                } else {
+                    splitSource[number] = line.replace(regex, '$1' + replacement + '$3');
+                }
+            }
+        }
+
+        function findInSource(str) {
```

Comment:
Usually we do not use inner functions, so make all these separate outer functions.